editors/code/src/logging.ts

```diff
@@ -0,0 +1,28 @@
+import * as util from "util";
+import { Disposable, OutputChannel } from "vscode";
+
+type Arguments = unknown[];
+class OutputChannelLogger {
+	constructor(private readonly channel: OutputChannel) {}
+
+	public outputLog(...data: Arguments): void {
+		this.channel.appendLine(util.format(...data));
+	}
+}
+
+let channel: OutputChannelLogger | undefined;
+export function registerLogger(logChannel: OutputChannel): Disposable {
+	channel = new OutputChannelLogger(logChannel);
+	return {
+		dispose: () => {
+			channel = undefined;
+		},
+	};
+}
+
+export function outputLog(...args: Arguments): void {
```

Comment:
Can the file be called `output` and this function `log()` so that we'd do `output.log()`?